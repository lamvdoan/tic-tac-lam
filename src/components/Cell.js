import React from 'react';
import { connect } from 'react-redux';
import { cellIsMarked, setIsGameActive, setCurrentPlayer } from '../actions/game';
import { O_CELL_IMAGE, X_CELL_IMAGE, EMPTY_CELL_IMAGE, IMAGE_HOME } from '../properties/imageFileSystem';

const imagePath = {
    emptyCell: `${IMAGE_HOME}/${EMPTY_CELL_IMAGE}`,
    oCell: `${IMAGE_HOME}/${O_CELL_IMAGE}`,
    xCell: `${IMAGE_HOME}/${X_CELL_IMAGE}`,
}

export class Cell extends React.Component {
    onClick = () => {
        console.log('Clicking cell: ' + this.props.id);
        this.props.cellIsMarked(this.props.id);
        this.props.setCurrentPlayer();

        // console.log('Grid Letter: ' + this.props.game.grid[this.props.id]);
        // console.log('Grid: ' + this.props.game.grid);
    };

    getImageName = () => {
        if (this.props.game.grid[this.props.id] === 'X') {
            return 'xCell';
        } else if (this.props.game.grid[this.props.id] === 'O') {
            return 'oCell';
        }
    }

    render() {
        const imageName = this.getImageName();
        return (
            <img onClick={this.onClick} className="cell" src={imagePath[imageName]} />
        );
    }
}

const mapStateToProps = (state) => ({
    game: state.game
});

const mapDispatchToProps = (dispatch) => ({
    cellIsMarked: (index) => dispatch(cellIsMarked(index)),
    setIsGameActive: () => dispatch(setIsGameActive()),
    setCurrentPlayer: () => dispatch(setCurrentPlayer())
});

export default connect(mapStateToProps, mapDispatchToProps)(Cell);
