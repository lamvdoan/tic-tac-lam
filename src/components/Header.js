import React from 'react';
import { Link } from 'react-router-dom';

const Header = () => (
    <header className="header">
        <div className="content-container">
            <Link className="header__title" to="/">
                <h1>Tic Tac Lam</h1>
            </Link>
        </div>
    </header>
);
export default Header;
