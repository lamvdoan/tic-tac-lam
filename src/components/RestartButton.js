import React from 'react';
import { connect } from 'react-redux';
import { restartGame } from '../actions/game';

export class RestartButton extends React.Component {
    onClick = () => {
        this.props.restartGame();
    }

    render() {
        return (
            <button onClick={this.onClick} className="button">Play Again</button>
        );
    }
}

const mapDispatchToProps = (dispatch) => ({
    restartGame: () => dispatch(restartGame())
});

export default connect(null, mapDispatchToProps)(RestartButton);
