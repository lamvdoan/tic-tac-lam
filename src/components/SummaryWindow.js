import React from 'react';
import { connect } from 'react-redux';

export const SummaryWindow = ({ currentPlayer }) => (
    <div>
        <p>Current Player: {currentPlayer}</p>
    </div>
);

const mapStateToProps = (state) => ({
    currentPlayer: state.currentPlayer
});

export default connect(mapStateToProps)(SummaryWindow);