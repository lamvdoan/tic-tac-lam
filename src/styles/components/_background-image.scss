.background-image {
    background: maroon url("/images/background.jpg") no-repeat center center fixed;
    background-size: 100%;
}