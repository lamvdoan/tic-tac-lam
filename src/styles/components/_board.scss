.board {
    margin: 0 0 0 $l-size;

    th {
        border: .1rem solid black;
    }
}
