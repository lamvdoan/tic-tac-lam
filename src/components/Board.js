import React from 'react';
import Cell from './Cell';

export class Board extends React.Component {
    render() {
        return (
            <div className="board">
                <div className="board__row">
                    <Cell id="0" />
                    <Cell id="1"/>
                    <Cell id="2"/>
                </div>
                <div className="board__row">
                    <Cell id="3"/>
                    <Cell id="4"/>
                    <Cell id="5"/>
                </div>
                <div className="board__row">
                    <Cell id="6"/>
                    <Cell id="7"/>
                    <Cell id="8"/>
                </div>
            </div>
        );
    }
}

export default Board;
