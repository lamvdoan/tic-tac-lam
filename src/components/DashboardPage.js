import React from 'react';
import Board from './Board';

const DashboardPage = () => (
    <div>
        <Board />
    </div>
);

export default DashboardPage;