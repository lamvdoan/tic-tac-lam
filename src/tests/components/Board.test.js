import React from 'react';
import { shallow } from 'enzyme';
import { Board } from '../../components/Board';

test('should move be valid', () => {
    const wrapper = shallow(<Board />);
    expect(wrapper).toMatchSnapshot();
});
