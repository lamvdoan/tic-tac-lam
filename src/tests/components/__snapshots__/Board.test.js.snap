// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should move be valid 1`] = `
<div
  className="board"
>
  <table>
    <tbody>
      <tr>
        <th>
          <Connect(Cell)
            id="0"
          />
        </th>
        <th>
          <Connect(Cell)
            id="1"
          />
        </th>
        <th>
          <Connect(Cell)
            id="2"
          />
        </th>
      </tr>
      <tr>
        <th>
          <Connect(Cell)
            id="3"
          />
        </th>
        <th>
          <Connect(Cell)
            id="4"
          />
        </th>
        <th>
          <Connect(Cell)
            id="5"
          />
        </th>
      </tr>
      <tr>
        <th>
          <Connect(Cell)
            id="6"
          />
        </th>
        <th>
          <Connect(Cell)
            id="7"
          />
        </th>
        <th>
          <Connect(Cell)
            id="8"
          />
        </th>
      </tr>
    </tbody>
  </table>
</div>
`;
